### PR TITLE
Feature/custom weather cache

### PR DESCRIPTION
**Pull request by daveyeager**

### CODE REVIEW

File: eemeter/weather/cache.py [hunk 1]
```diff
@@ -1,97 +1,81 @@
...
+        return 'SqlJSONStore("{}")'.format(self.url)
 
-    def _prepare_db(self, directory=None):
+    def _get_url(self):
+        url = os.environ.get("EEMETER_WEATHER_CACHE_URL")
+        if url is None:
+            directory = "{}/.eemeter/cache".format(os.path.expanduser('~'))
+            if not os.path.exists(directory):
+                os.makedirs(directory)
+            url = "sqlite:///{}/weather_cache.db".format(directory)
```
Comment (philngo) [inline eemeter/weather/cache.py]: Did you consider making the postgres database the default backend? I wonder the benefits of retaining the weather store would outweigh the risks of interfering with the django tables.

File: tests/ee/test_energy_efficiency_meter_trace_centric.py [hunk 2]
```diff
@@ -92,16 +92,16 @@ def meter_input_strange_interpretation():
 
 @pytest.fixture
 def mock_isd_weather_source():
```
Comment (philngo) [inline tests/ee/test_energy_efficiency_meter_trace_centric.py]: On a future refactor this is a good candidate for being made into a factory.

File: eemeter/weather/cache.py (hunk 1 above)
Comment (daveyeager) [inline eemeter/weather/cache.py]: I did - reason I decided against it was I wanted the library to still be as portable as possible. Whether that was a good reason or not we can certainly discuss.

**Comment by marcpareoee** [comment]
Looks great overall. 

One thing that I wonder about is how to handle migrations in the cache. I don't think the implementation handles the table structure changing? Maybe I didn't look closely enough.

The simple approach is to blow away the cache if the table structure has changed. But maybe this isn't desirable if it's expensive to rebuild the cache.

Another approach is to write migration SQL statements that are idempotent. e.g. CREATE COLUMN IF NOT EXISTS. That might end up working well enough to not need a more heavyweight migrations solution like Alembic.

Might not even need to add anything to the implementation for now...just wanted to make sure it is thought through ahead of time.

***

About a Postgres deployment. How about a new manifest in Chief with just a database? Maybe a simple Flask app on the `web` dyno with some diagnostics. Is there any reason it would be useful to have a script to populate the cache rather than rely on eemeter? That could live in this app, too.

**Comment by philngo** [comment]
Merging this. I think blowing away the cache is probably the right way to handle the table structure changes for now, especially since we don't anticipate many. It may be expensive to rebuild it once it grows, but it won't be any worse than what is currently done - blowing away the cache for every individual instance - and can be done incrementally without any performance cost to the end user.

**Comment by philngo** [comment]
Oops, didn't mean to close

**Pull request merged**